test/stableX_large.js

```diff
@@ -0,0 +1,183 @@
+} = require("./utilities.js")
+
+const feeDenominator = 1000 // fee is (1 / feeDenominator)
+
+function feeSubtracted(x) {
+  return Math.floor(x * (feeDenominator - 1) / feeDenominator)
+}
+
+function feeAdded(x) {
+  return Math.floor(x * (feeDenominator) / (feeDenominator - 1))
+}
+
+contract("StablecoinConverter", async (accounts) => {
+
+  const [user_1, user_2, solutionSubmitter] = accounts
+
+  // Basic Trade used in most of the tests:
+  // Trade for user_1: amount of token_1 sold: 20020, amount of token_2 bought: 10000,
+  // Trade for user_2: amount of token_2 sold: 10000, amount of token_1 bought: feeSubtracted(10000) * 2
+  // ==> Token conservation holds for token_2, and fee token == token_1 has negative balance of 40
+
+  const basicTrade = {
+    deposits: [{ amount: feeAdded(2000000), token: 0, user: user_1 }, { amount: feeAdded(1000000) * 2, token: 1, user: user_2 }],
+    orders: [
+      { sellToken: 0, buyToken: 1, sellAmount: feeAdded(2000000), buyAmount: 1000000, user: user_1 },
+      { sellToken: 1, buyToken: 0, sellAmount: feeAdded(1000000), buyAmount: feeSubtracted(1000000) * 2, user: user_2 }
+    ],
+    solution: { prices: [1, 2], owners: [user_1, user_2], volume: [1000000, feeSubtracted(1000000) * 2], tokenIdsForPrice: [0, 1] }
+  }
+
+  let stablecoinConverter
+  let batchIndex, orderId1, orderId2, prices, seedOwners, seedVolumes, tokenIdsForPrice
+  beforeEach(async () => {
+    const feeToken = await MockContract.new()
+    stablecoinConverter = await StablecoinConverter.new(2 ** 16 - 1, feeDenominator, feeToken.address)
+
+    const erc20_2 = await MockContract.new()
+    await feeToken.givenAnyReturnBool(true)
+    await erc20_2.givenAnyReturnBool(true)
+
+    await stablecoinConverter.deposit(feeToken.address, basicTrade.deposits[0].amount, { from: basicTrade.deposits[0].user })
+    await stablecoinConverter.deposit(erc20_2.address, basicTrade.deposits[1].amount, { from: basicTrade.deposits[1].user })
+
+    await stablecoinConverter.addToken(erc20_2.address)
+    batchIndex = (await stablecoinConverter.getCurrentBatchId.call()).toNumber()
+
+    orderId1 = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, basicTrade.orders[0].buyToken, basicTrade.orders[0].sellToken, true, batchIndex + 1, basicTrade.orders[0].buyAmount, basicTrade.orders[0].sellAmount, { from: basicTrade.orders[0].user })
+    orderId2 = await sendTxAndGetReturnValue(stablecoinConverter.placeOrder, basicTrade.orders[1].buyToken, basicTrade.orders[1].sellToken, true, batchIndex + 1, basicTrade.orders[1].buyAmount, basicTrade.orders[1].sellAmount, { from: basicTrade.orders[1].user })
+
+    await closeAuction(stablecoinConverter)
+
+    prices = basicTrade.solution.prices
+    seedOwners = basicTrade.solution.owners
+    seedVolumes = [10000, feeSubtracted(10000) * 2]
+    tokenIdsForPrice = basicTrade.solution.tokenIdsForPrice
+  })
+
+  describe("submit single solutions", () => {
+    it("filling 10 orders", async () => {
```

Comment:
Would we be able to reduce a lot of repetition here if we created a function fill_n_orders and called it with 10, 20 & 40 in each case?

Reply:
Already working on it